typescript/web/heap.ts

```diff
@@ -1,36 +1,49 @@
-let heapDestinationInterface = {
-  make: function (env: AvoEnv, apiKey: string) {
-    window.heap.load(apiKey, {
-      disableTextCapture: true
-    })
-  },
-  logEvent: function (eventName, eventProperties) {
-    window.heap.track(
-      eventName,
-      eventProperties as Record<string, unknown>
-    )
-  },
-  setUserProperties: function (userId, userProperties) {
-    window.heap.addUserProperties(
-      userProperties as Record<string, unknown>
-    )
-  },
-  identify: function (userId) {
-    window.heap.identify(userId)
-  },
-  unidentify: function () {
-    window.heap.resetIdentity()
-  },
-  logPage: function (pageName: string, eventProperties: object) {
-    window.heap.track(
-      "page",
-      eventProperties.assign({ "pageName": pageName }) as Record<string, unknown>
-    )
-  },
-  revenue: function (amount: number, eventProperties: object) {
-    window.heap.track(
-      "revenue",
-      eventProperties.assign({ "revenue": amount })
-    )
-  },
-}
\ No newline at end of file
+// Install Heap following their guide: https://developers.heap.io/docs/web
+// Heap should now be available as window.heap
+
+// Import the Avo Codegen file from the location you've chosen
+import { CustomDestination } from "../path/to/Avo";
+
+// Get the types from here: https://developers.heap.io/reference/client-side-apis-overview
+interface Heap {
+  load: (appId: string, options?: Object) => void; // For some reason, the load function is missing from the docs so we're adding it
+  track: (event: string, properties?: Object) => void;
+  identify: (identity: string) => void;
+  resetIdentity: () => void;
+  addUserProperties: (properties: Object) => void;
+  addEventProperties: (properties: Object) => void;
+  removeEventProperty: (property: string) => void;
+  clearEventProperties: () => void;
+  appid: string;
+  userId: string;
+  identity: string | null;
+  config: any;
+}
+
+// Additionally we'll need to let TypeScript know that Heap is available as window.Heap
+declare global {
+  interface Window {
+    heap: Heap;
+  }
+}
+
+const heapDestinationInterface: CustomDestination = {
+  make: (_env, apiKey) =>
+    window.heap.load(apiKey, { disableTextCapture: true }),
+
+  identify: (userId) => window.heap.identify(userId),
+
+  logEvent: (eventName, eventProperties) =>
+    window.heap.track(eventName, eventProperties),
+
+  setUserProperties: (_userId, userProperties) =>
+    window.heap.addUserProperties(userProperties),
+
+  unidentify: () => window.heap.resetIdentity(),
+
+  logPage: (pageName, eventProperties) =>
+    window.heap.track("page", { ...eventProperties, pageName }),
+
+  revenue: (amount, eventProperties) =>
+    window.heap.track("revenue", { ...eventProperties, revenue: amount }),
```

Comment:
I am not finding in their docs about having heap.track("revenue"...) or heap.track("page", ...) for these events, 100% sure this is how to track revenue and page events in heap? can we add link to the docs for these above the function?

Reply:
Great question! The `heap.track` function takes in the event name as the first parameter so what's effectively happening is that we're just creating a new event called "revenue" and attributing the amount to that. I'll add a comment about that

Reply:
Added a comment! What do you think?